contracts/registry/implementation/IdentityRegistryStorageERC3643.sol

```diff
@@ -0,0 +1,173 @@
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+pragma solidity 0.8.27;
+
+import "../../errors/InvalidArgumentErrors.sol";
+
+import "../../roles/AgentRoleUpgradeable.sol";
+import "../../roles/IERC173.sol";
+import "../interface/IIdentityRegistryStorage.sol";
+import "../storage/IRSStorage.sol";
+import "@onchain-id/solidity/contracts/factory/IIdFactory.sol";
+import "@onchain-id/solidity/contracts/interface/IIdentity.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+/// Errors
+
+/// @dev Thrown when address is already stored
+error AddressAlreadyStored();
+
+/// @dev Thrown when address is not yet stored.
+error AddressNotYetStored();
+
+/// @dev Thrown when trying to call write functions.
+error ContractIsReadOnly();
+
+/// @dev Thrown when maximum numbe of identity registry by identity registry storage is reached.
+/// @param _max miximum number of IR by IRS.
+error MaxIRByIRSReached(uint256 _max);
+
+
+contract IdentityRegistryStorage is IERC3643IdentityRegistryStorage, AgentRoleUpgradeable, IRSStorage, IERC165 {
+    IIdFactory internal _iidFactory;
+    function init(_identityFactoryAddress) external initializer {
+        require(_identityFactoryAddress != address(0), ZeroAddress());
+        _iidFactory = IIdFactory(_identityFactoryAddress);
+        __Ownable_init();
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-addIdentityToStorage}.
+     */
+    function addIdentityToStorage() external view override onlyAgent {
+        revert ContractIsReadOnly();
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-modifyStoredIdentity}.
+     */
+    function modifyStoredIdentity() external view override onlyAgent {
+        revert ContractIsReadOnly();
+
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-modifyStoredInvestorCountry}.
+     */
+    function modifyStoredInvestorCountry() external view override onlyAgent {
+        revert ContractIsReadOnly();
+
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-removeIdentityFromStorage}.
+     */
+    function removeIdentityFromStorage() external view override onlyAgent {
+        revert ContractIsReadOnly();
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-bindIdentityRegistry}.
+     */
+    function bindIdentityRegistry() external view override {
+        revert ContractIsReadOnly();
```

Comment:
just do nothing, don't revert, otherwise the factory won't be able to use this contract, as the factory will try to bind